zap/src/main/java/org/zaproxy/zap/extension/ascan/ScanPolicy.java

```diff
@@ -114,23 +121,30 @@ public void save() throws ConfigurationException {
         this.conf.save();
     }
 
-    private <T extends Enum<T>> T getEnumFromConfig(
-            String confKey, Class<T> enumType, T defaultValue) {
-        String name = conf.getString(confKey, "");
-        if (name.isEmpty()) {
-            return defaultValue;
+    private AlertThreshold getAlertThresholdFromConfig() {
+        String alertThreshold = conf.getString("scanner.level", "");
+        if (alertThreshold.isEmpty()
+                || !EnumUtils.isValidEnum(AlertThreshold.class, alertThreshold)
+                || AlertThreshold.OFF.name().equals(alertThreshold)
+                || AlertThreshold.DEFAULT.name().equals(alertThreshold)) {
+            logger.warn(
+                    "Found illegal value {} for alert threshold, using default (MEDIUM) instead.",
```

Comment:
The user can set "default" to any of the regular values, so this is a bit misleading. Perhaps:
```suggestion
                    "Found illegal value {} for alert threshold, using MEDIUM instead.",
```

Reply:
Something similar for Str too?

Reply:
Thanks, tweaked.